### PR TITLE
Initiate earthly-mode

### PR DESCRIPTION
**Pull request by wingyplus**
This is my work from https://github.com/wingyplus/earthly-mode. The current features of the major mode are:

* Syntax highlight support
* 2 file extension support, `Earthfile` and `.earth`.

I need to special thanks to @TheCoreMan, who initiate the syntax highlight, and @duck1123, who help us test the syntax highlighter. :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,50 @@
-# earthly-mode
\ No newline at end of file
+# earthly-mode
+
+Major mode for editing Earthly Earthfile (https://docs.earthly.dev/docs/earthfile)
```
Comment (vladaionescu) [inline README.md:3]:
```suggestion
Major mode for editing Earthly Earthfile (https://docs.earthly.dev/docs/earthfile) developed by @wingyplus, @TheCoreMan and @duck1123
```

File: README.md [hunk 2]
```diff
@@ -1 +1,50 @@
...
+
+Or if you use `use-package`, just add:
+
+```elisp
+(use-package earthly-mode
+  :ensure t
+  :mode (("\\.earth\\'" . earthly-mode)
+         ("\\Earthfile\\'" . earthly-mode)))
+```
+
```
Comment (vladaionescu) [inline README.md:47]: This is no longer necessary, correct?

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+.DS_Store
```
Comment (vladaionescu) [inline .gitignore:1]: Nit: We generally prefer to not put this in the project's `.gitignore` and instead leave it to each user to define these in their global git ignore.

**Review by vladaionescu — COMMENTED** [review]
Looks great! Some small suggestions...

File: .gitignore (hunk 3 above)
Comment (wingyplus) [reply .gitignore]: I remove this line out of `.gitignore`. :)

File: README.md (hunk 2 above)
Comment (wingyplus) [reply README.md]: You're right. It's remove in https://github.com/earthly/earthly-mode/pull/1/commits/dcbdeb306b5fe9b2fa231b22051d519470cd5f96.

File: README.md (hunk 1 above)
Comment (wingyplus) [reply README.md]: Added in https://github.com/earthly/earthly-mode/pull/1/commits/dcbdeb306b5fe9b2fa231b22051d519470cd5f96.

**Comment by wingyplus** [comment]
@vladaionescu Thanks for reviewing this PR. Your suggestion has already been addressed by https://github.com/earthly/earthly-mode/pull/1/commits/dcbdeb306b5fe9b2fa231b22051d519470cd5f96 and https://github.com/earthly/earthly-mode/pull/1/commits/0503b8e53f19d6713b041f17a0fec73967f32337. :)

**Review by vladaionescu — APPROVED** [review]

**Comment by wingyplus** [comment]
🎉 🎉

**Pull request merged**